<?xml version="1.0"?>
<launch>
    <arg name="map_frame_id" default="map"/>
    <arg name="base_link_frame_id" default="base_foot_point"/>
    <arg name="loop_rate" default="50"/>
    <arg name="fileName" default="way1"/>

    <include file="$(find tvc_navigation)/launch/include/slam_2d.launch">
        <arg name="nav_mode" default="slam"/>
        <arg name="mapFile" value="$(find tvc_navigation)/resources/map_2d/tukuba/map.yaml"/>
    </include>

    <!--joystick-->
    <node name="joystickPS4" pkg="tvc_navigation" type="joystickPS4">
        <param name="control_rate" value="1.0"/>
    </node>
    <node name="joy_node" pkg="joy" type="joy_node"/>

    <node name="wpRecord" pkg="tvc_wpTool" type="wpRecord">
        <param name="map_frame_id" value="$(arg map_frame_id)"/>
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)"/>
        <param name="loop_rate" value="$(arg loop_rate)"/>
        <param name="wayPoint_pitch" value="0.5"/>
    </node>

    <node name="csv_output" pkg="tvc_wpTool" type="csv_output">
        <param name="filePath" value="$(find tvc_navigation)/csv/$(arg fileName)"/>
        <remap from="csv_array"  to="path2csv"/>
    </node>

    <!--rviz-->
    <arg name="rvizconfig" default="$(find tvc_navigation)/rviz/wpRecord.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" respawn="true" />
    
</launch>